### PR TITLE
Added leetcode problem 10

### PR DESCRIPTION
**Pull request by PythonicBoat**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by PythonicBoat** [comment]
@Panquesito7 can you please review my PR?

File: leetcode/src/10.c [hunk 1]
```diff
@@ -0,0 +1,25 @@
+bool isMatch(char* s, char* p);
```
Comment (Panquesito7) [inline leetcode/src/10.c:1]: Could you please add small documentation about the problem?

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please resolve the merge conflicts (for the `leetcode/README.md` file) [**here**](https://github.com/TheAlgorithms/C/pull/1033/conflicts).
Let us know if you need any help here or on our [**Discord server**](https://the-algorithms.com/discord)! 🙂

File: leetcode/src/10.c (hunk 1 above)
Comment (PythonicBoat) [reply leetcode/src/10.c:16]: added documetation

File: leetcode/src/10.c [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/* Prompt :
```
Comment (Panquesito7) [inline leetcode/src/10.c:1]:
```suggestion
/*
Prompt:
```

File: leetcode/src/10.c [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+The matching should cover the entire input string (not partial).
+
+Constraints:
+
+1 <= s.length <= 20
+1 <= p.length <= 30
+s contains only lowercase English letters.
+p contains only lowercase English letters, '.', and '*'.
+It is guaranteed for each appearance of the character '*', there will be a previous valid character to match.
+*/
```
Comment (Panquesito7) [inline leetcode/src/10.c:15]:
```suggestion

It is guaranteed for each appearance of the character '*', there will be a previous valid character to match.
*/

```

**Review by Panquesito7 — APPROVED** [review]
Thanks! 🚀

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]
I think you've forgotten to write tests for this implementation... EDIT: Seems that none of the algorithms do. I'll approve it in that case.

**Review by tjgurwara99 — APPROVED** [review]

**Review by Panquesito7 — APPROVED** [review]

**Pull request merged**